<?xml version="1.0"?>
<launch>

  <arg name="no_static_map" default="true"/>

  <arg name="base_global_planner" default="navfn/NavfnROS"/>
  <arg name="base_local_planner" default="teb_local_planner/TebLocalPlannerROS"/>
  <!-- <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/> -->

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <remap from="cmd_vel" to="cmd_vel_neg" />
    <remap from="odometry/filtered" to="odometry/filtered_inv" />
    <remap from="move_base_simple/goal" to="move_base_simple_inverted/goal" />
    <param name="base_global_planner" value="$(arg base_global_planner)"/> 
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <rosparam file="$(find igvc_navigation)/config/planner.yaml" command="load"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find igvc_navigation)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find igvc_navigation)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find igvc_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>

    <!-- static global costmap, static map provides size -->
    <rosparam file="$(find igvc_navigation)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>
    
    <!-- global costmap with laser, for odom_navigation_demo -->
    <rosparam file="$(find igvc_navigation)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
    <param name="global_costmap/width" value="100.0" if="$(arg no_static_map)"/>
    <param name="global_costmap/height" value="100.0" if="$(arg no_static_map)"/>
  </node>
  <node pkg="tf" type="static_transform_publisher" name="velodyne_broadcaster" args="0 0 1 0 0 0 1 base_link velodyne 100" />
  
  <node pkg="igvc_navigation" type = "odom_inv" name = "inver_odom" />
  <node pkg="igvc_navigation" type="vel_inv" name="inver_vel" />


</launch>
